### DISCUSSION TITLE
In Microsoft Search data source, with people entity the results are not accurate.

### DISCUSSION THREAD

**Discussion by GangadharMFord** [microsoft-search/pnp-modern-search#3905] [Q&A] [answered]
Hi, I'm using PnP modern search v4 4.8 version for my search results. recently we are migrating the search data source from SharePoint online index to Microsoft search data source with results. We have found that in people section (people entity - MS data source) the search results for input query is not same or accurate as sharepoint online (local people results) or office.com people results. 
Microsoft search people entity is returning very minimal comparing with SPO index and office.com.
As part of my troubleshoot the issue check the api permissions and it is People.read and People.Read.all permissions too.

I checked the same with using graph explorer for people entity and results are same (very minimal results are returned). 

Can you advise is this limitation or bug in Microsoft Search People entity / graph api call ?  Or something we missed?

**Comment by kasperbolarsen** [comment, marked as answer]
As per https://learn.microsoft.com/en-us/graph/search-concept-person we only get "my people" from the API, not the entire dataset.
I am working on getting directions from the MS Search team how and when we will be able to get the complete dataset, as each and every MS People search employee directory will fall flat on its face otherwise.
For the time being I have the recommend that everybody uses SharePoint People Search rather than MS Search :-(

**Comment by GangadharMFord** [comment]
> As per https://learn.microsoft.com/en-us/graph/search-concept-person we only get "my people" from the API, not the entire dataset. I am working on getting directions from the MS Search team how and when we will be able to get the complete dataset, as each and every MS People search employee directory will fall flat on its face otherwise. For the time being I have the recommend that everybody uses SharePoint People Search rather than MS Search :-(

Thank you Kasper with your above input's. is there any other alternatives from Microsoft Search graph api to fetch the complete dataset for query. I have also tested using people end point and still no luck.

**Comment by kasperbolarsen** [comment]
Unfortunately not.  I'll revert as soon as I hear from the product team
